### PR TITLE
makes fire overlay onto walls like fluids

### PR DESCRIPTION
**Pull request by Skeletonman0**
[ATMOSPHERICS] [QOL] [SPRITES]
## About the PR 
adds some code onto obj/hotspot borrowed from fluids to make fireflashes / plasmafires overlay onto walls
this does mean that it is visible from the other side of the wall however

i thought about applying this to atmos gases themselves, but that would probably be way too difficult
due to how they work in the first place.

## Why's this needed? 
looks nicer
image

### CODE REVIEW

**Comment by virvatuli-gh** [comment]
:eyes:

File: code/modules/atmospherics/FEA_fire.dm [hunk 1]
```diff
@@ -185,7 +186,25 @@
 #else
 			add_medium_light("hotspot", list(red, green, blue, 100))
 #endif
-
+	set_loc() // hotspots apparently start somewhere wacky
+		..()
+		src.update_overlay_blocking()
+		src.update_perspective_overlays()
```
Comment (ZeWaka) [inline code/modules/atmospherics/FEA_fire.dm:192]: you should surround this and check if the loc is null, no use doing this stuff if we're in nullspace - that should resolve the !t issue below

**Comment by Glamurio** [comment]
I think it's a really slick effect, but I personally don't like that it's visible from the other side of the wall, though you win some and you lose some I guess

**Comment by TobleroneSwordfish** [comment]
Might be worth profiling?

**Comment by ZeWaka** [comment]
it doesn't need to be visible from the other side of the wall, offset overlays

**Review by ZeWaka — CHANGES_REQUESTED** [review]
no changes necessarily requested, but I'd like a performance evaluation of how expensive this is - via the disco inferno scenario()

**Comment by pali6** [comment]
> I think it's a really slick effect, but I personally don't like that it's visible from the other side of the wall, though you win some and you lose some I guess

That's actually a one line fix! :)

File: code/modules/atmospherics/FEA_fire.dm [hunk 2]
```diff
@@ -272,14 +293,72 @@
 				for(var/turf/simulated/possible_target in possible_spread)
 					if(!possible_target.active_hotspot)
 						possible_target.hotspot_expose(radiated_temperature, CELL_VOLUME/4)
+						src.update_overlay_blocking()
+						src.update_perspective_overlays()
```
Comment (pali6) [inline code/modules/atmospherics/FEA_fire.dm:297]: What's the reasooning behind thse calls here?

**Comment by pali6** [comment]
image
some corners have overlapping overlays which looks kinda bad

**Comment by pali6** [comment]
image
Walls getting deleted doesn't quite update the overlays properly.

**Comment by pali6** [comment]
image
Windows don't get any overlays when cardinally attached to a fire which looks kinda bad.

**Review by pali6 — CHANGES_REQUESTED** [review]
See my previous comments. Basically this will need to hook somewhere into the `tilenotify` proc on turfs most likely to keep the overlays up to date with wall changes. You will also need slightly different conditions for when to place overlays than fluids have (for example place these on windows). Also you will need to make it so if a has a fire both to the south and to east/west its bottom corners don't get covered by two overlays.

Also, for performance reasons, `blocked_perspective_objects` could be a flag bit field instead of an array. And `wall_overlay_images` could be a `static` where you give it a direction and icon state and it'll give you the right overlay image, no need to have it be on each hotspot.

**Comment by ZeWaka** [comment]
zewaka idea: don't use overlays, just make the fire icon like 50x50 and use icon states for cheaper

**Pull request closed without merging**